src/arch/linux/mapping/mapping.c

```diff
@@ -1065,14 +1068,16 @@ int alias_mapping_pa(int cap, unsigned addr, size_t mapsize, int protect,
   void *addr2;
   struct hardware_ram *hw;
   dosaddr_t va = do_get_hardware_ram(addr, mapsize, &hw);
-  if (va == (dosaddr_t)-1)
+  if (hw == NULL)
     return 0;
   assert(addr >= LOWMEM_SIZE + HMASIZE);
+  hwram_update_aliasmap(hw, addr, mapsize, source);
+  if (va == (dosaddr_t)-1)
```

Comment:
So here's the trick, but the patch
that moves a call out of config.dpmi
check goes before?
Looks clearly wrong.
You probably should move that "move"
to this patch.

Reply:
I tested the patch before. `alias_mapping_pa` works ok in the first patch much like it does for XMS, because after patch 1 the mem_base area still exists and is registered as vbase for !config.dpmi.

Reply:
But the log of the first patch says otherwise:
````This way with $_dpmi=(0) virtual mem_base addresses >= 0x110000 are
no longer accessed anywhere.```

Reply:
I tried to reword this now. When config.dpmi==0 that memory is still mapped but not accessed. It can't be accessed by regular DOS code since DPMI is disabled and VCPI not merged yet.

The only place that accessed it was `int15()` via `physaddr_to_unixaddr()`.

So after this change that mem_base area is mapped but unused. In the 6th (last) patch in this PR I no longer map it either (which needs the hardware ram with vbase==-1), but in the 3rd patch I already need alias_mapping_pa for extmem to map both virtual and physical memory to shm.